### PR TITLE
feature: Capture important events with span in fission and add trace id in logs

### PR DESCRIPTION
**Pull request by sanketsudake**
Signed-off-by: Sanket Sudake <sanketsudake@gmail.com>

1. We capture all spans from router to function pod
image

2. Within a span different events are captured
image

3. Logs with context should have traceID, so by trace ID we can travel all logs. Logs with trace ID are enabled even if opentelemtry collector not mentioned.
```text
executor-7758b648f6-2tsgn executor {"level":"info","ts":"2021-09-14T11:32:01.971Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/gp.go:480","msg":"chose pod","trace_id":"26394b2c0739b896b5dada41d2bba00e","labels":{"environmentName":"go","environmentNamespace":"default","environmentUid":"97e63265-4846-4b19-8035-d6d862164409","executorType":"poolmgr","functionName":"go-otel-func","functionNamespace":"default","functionUid":"53c273ec-842c-4071-bd62-13d944a251cf","managed":"false"},"pod":"poolmgr-go-default-2129733-8f85f8f65-kpg82","elapsed_time":0.0274459}
executor-7758b648f6-2tsgn executor {"level":"info","ts":"2021-09-14T11:32:01.971Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/gp.go:485","msg":"calling fetcher to copy function","trace_id":"26394b2c0739b896b5dada41d2bba00e","function":"go-otel-func","url":"http://10.244.0.73:8000/"}
executor-7758b648f6-2tsgn executor {"level":"info","ts":"2021-09-14T11:32:01.971Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/gp.go:485","msg":"specializing pod","trace_id":"26394b2c0739b896b5dada41d2bba00e","function":"go-otel-func"}
executor-7758b648f6-2tsgn executor {"level":"info","ts":"2021-09-14T11:32:02.184Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/gpm.go:197","msg":"specialized pod","trace_id":"26394b2c0739b896b5dada41d2bba00e","function":"go-otel-func","functionNamespace":"default","env":"go","envNamespace":"default","pod":"poolmgr-go-default-2129733-8f85f8f65-kpg82","podNamespace":"fission-function","podIP":"10.244.0.73"}
executor-7758b648f6-2tsgn executor {"level":"info","ts":"2021-09-14T11:32:02.220Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/gpm.go:197","msg":"added function service","trace_id":"26394b2c0739b896b5dada41d2bba00e","function":"go-otel-func","functionNamespace":"default","env":"go","envNamespace":"default","pod":"poolmgr-go-default-2129733-8f85f8f65-kpg82","podNamespace":"fission-function","serviceHost":"10.244.0.73:8888","podIP":"10.244.0.73"}
executor-7758b648f6-2tsgn executor {"level":"info","ts":"2021-09-14T11:34:04.971Z","logger":"generic_pool_manager","caller":"runtime/asm_amd64.s:1371","msg":"release idle function resources","function":"go-otel-func","address":"10.244.0.73:8888","executor":"poolmgr","pod":"poolmgr-go-default-2129733-8f85f8f65-kpg82"}
```

### CODE REVIEW

File: pkg/utils/otel/attributes.go [hunk 1]
```diff
@@ -1,24 +1,133 @@
...
+	if deployment == nil {
+		return []attribute.KeyValue{}
+	}
+	return []attribute.KeyValue{
+		{Key: "deployment-name", Value: attribute.StringValue(deployment.Name)},
+		{Key: "deployment-namespace", Value: attribute.StringValue(deployment.Namespace)},
+	}
+}
+
+func GetAttributesForHpa(hpa *asv1.HorizontalPodAutoscaler) []attribute.KeyValue {
```
Comment (gauravgahlot) [inline pkg/utils/otel/attributes.go:99]:
```suggestion
func GetAttributesForHPA(hpa *asv1.HorizontalPodAutoscaler) []attribute.KeyValue {
```

**Review by gauravgahlot — CHANGES_REQUESTED** [review]
Just a nitpick. Rest looks good.

**Review by gauravgahlot — APPROVED** [review]
It would be helpful if we can add a image/screenshot of how the spans look now. Other than that, LGTM.

**Comment by sanketsudake** [comment]
Added screentshots and supporting logs

**Pull request merged**